File: servers/scalatra/src/main/scala/com/wordnik/client/model/Execution.scala
package com.wordnik.client.model

import java.util.Date
case class Execution (
  execID: String,

  orderID: Option[String],

  clOrdID: Option[String],

  account: Option[Double],

  symbol: Option[String],

  side: Option[String],

  lastQty: Option[Double],

  lastPx: Option[Double],

  underlyingLastPx: Option[Double],

  lastMkt: Option[String],

  lastLiquidityInd: Option[String],

  simpleOrderQty: Option[Double],

  orderQty: Option[Double],

  price: Option[Double],

  minQty: Option[Double],

  stopPx: Option[Double],

  currency: Option[String],

  settlCurrency: Option[String],

  execType: Option[String],

  ordType: Option[String],

  timeInForce: Option[String],

  exDestination: Option[String],

  ordStatus: Option[String],

  workingIndicator: Option[Boolean],

  ordRejReason: Option[String],

  simpleLeavesQty: Option[Double],

  leavesQty: Option[Double],

  simpleCumQty: Option[Double],

  cumQty: Option[Double],

  avgPx: Option[Double],

  commission: Option[Double],

  commType: Option[String],

  tradePublishIndicator: Option[String],

  multiLegReportingType: Option[String],

  text: Option[String],

  trdMatchID: Option[String],

  execCost: Option[Double],

  execComm: Option[Double],

  homeNotional: Option[Double],

  foreignNotional: Option[Double],

  transactTime: Option[Date],

  timestamp: Option[Date]

  )

File: clients/typescript-angular/API/Client/Execution.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    export interface Execution {

        execID: string;

        orderID?: string;

        clOrdID?: string;

        clOrdLinkID?: string;

        account?: number;

        symbol?: string;

        side?: string;

        lastQty?: number;

        lastPx?: number;

        underlyingLastPx?: number;

        lastMkt?: string;

        lastLiquidityInd?: string;

        simpleOrderQty?: number;

        orderQty?: number;

        price?: number;

        displayQty?: number;

        stopPx?: number;

        pegOffsetValue?: number;

        pegPriceType?: string;

        currency?: string;

        settlCurrency?: string;

        execType?: string;

        ordType?: string;

        timeInForce?: string;

        execInst?: string;

        contingencyType?: string;

        exDestination?: string;

        ordStatus?: string;

        triggered?: string;

        workingIndicator?: boolean;

        ordRejReason?: string;

        simpleLeavesQty?: number;

        leavesQty?: number;

        simpleCumQty?: number;

        cumQty?: number;

        avgPx?: number;

        commission?: number;

        tradePublishIndicator?: string;

        multiLegReportingType?: string;

        text?: string;

        trdMatchID?: string;

        execCost?: number;

        execComm?: number;

        homeNotional?: number;

        foreignNotional?: number;

        transactTime?: date;

        timestamp?: date;
    }

}
